Manager/TransUnitManager.php

```diff
@@ -122,7 +122,9 @@ public function updateTranslation(TransUnit $transUnit, $locale, $content, $flus
         $found = false;
 
         while ($i<$end && !$found) {
-            $found = ($transUnit->getTranslations()->get($i)->getLocale() == $locale);
+            if (null !== $transUnit->getTranslations()->get($i)) {
+                $found = ($transUnit->getTranslations()->get($i)->getLocale() == $locale);
```

Comment:
I'm not sure of what you are doing next but you surely can break the loop after you found a match.